WooCommerce/src/main/kotlin/com/woocommerce/android/ui/bookings/filter/BookingFilterListScreen.kt

```diff
@@ -66,62 +65,94 @@ fun BookingFilterListScreen(state: BookingFilterListUiState) {
         },
         containerColor = MaterialTheme.colorScheme.surface,
     ) { innerPadding ->
-        AnimatedContent(
-            targetState = state.currentPage,
-            transitionSpec = {
-                if (targetState is BookingFilterPage.List) {
-                    slideOut()
-                } else {
-                    slideIn()
-                }
-            },
-            label = "BookingFiltersAnimatedContent",
+        FiltersNavHost(
+            state = state,
             modifier = Modifier
                 .fillMaxSize()
                 .padding(innerPadding)
-        ) { page ->
-            when (page) {
-                is BookingFilterPage.List -> {
-                    BookingFilterRootPage(state.items)
-                }
+        )
 
-                BookingFilterPage.AttendanceStatus,
-                BookingFilterPage.BookingType,
-                BookingFilterPage.Customer,
-                BookingFilterPage.Location,
-                BookingFilterPage.PaymentStatus,
-                BookingFilterPage.ServiceEvent,
-                BookingFilterPage.TeamMember,
-                is BookingFilterPage.DateTime -> {
-                    DateTimeFilterPicker()
-                }
+        // The navigation is driven by the state, so we handle back navigation by calling onClose
+        // We need to ensure that this called after NavHost to make sure we receive back events
+        BackHandler {
+            state.onClose()
+        }
+    }
+}
+
+@Composable
+private fun FiltersNavHost(
+    state: BookingFilterListUiState,
+    modifier: Modifier
+) {
+    val navController = rememberNavController()
+
+    LaunchedEffect(state.currentPage) {
+        if (state.currentPage != BookingFilterPage.List) {
+            navController.navigate(state.currentPage.route) {
+                popUpTo(BookingFilterPage.List.route)
             }
+        } else {
+            navController.popBackStack(BookingFilterPage.List.route, false)
+        }
+    }
+
+    NavHost(
+        navController = navController,
+        startDestination = BookingFilterPage.List.route,
+        enterTransition = { slideIn(popNavigation = true) },
+        exitTransition = { slideOut(popNavigation = true) },
+        popEnterTransition = { slideIn(popNavigation = false) },
+        popExitTransition = { slideOut(popNavigation = false) },
+        modifier = modifier
+    ) {
+        composable(BookingFilterPage.List.route) {
+            BookingFilterRootPage(state.items)
+        }
```

Comment:
```suggestion
        composable<BookingFilterPage.List> {
            BookingFilterRootPage(state.items)
        }
```

np: We could use the type-safe destinations to build the graph. We need to make those serializable, but I don't think it's a problem here.

Reply:
Yes, I thought about, but given we don't use the serialization plugin on the app, I didn't use, and given we most likely won't need it as the state already exposes all the information that we need, I decided to just use String routes, and they will be static routes.

Reply:
Related to this, I thought about converting `BookingFilterPage` to an enum, but I'm not sure if we'll need to pass data for the items in the future, I think given the root ViewModel state is available, we probably won't need any addiitonal data on the `currentPage` property, WDYT?
If we use an item, we can then just use the `name` property for route, but it's a minor point.

Reply:
Good call. At the moment, I can't think of any data we would pass so I think we should be okay with an enum!

Reply:
Done in 6474dc8, I kept using Camel Case for the item names, and kept the `route` extension, this way even if we need to use a `sealed inteface`, it will be an easy change.